frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts

```diff
@@ -324,13 +324,27 @@ export class ExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
 
   validateHasConditionCodeDefault(conditions: ExperimentCondition[]) {
     let defaultKeyword = this.translate.instant('home.new-experiment.design.condition.invalid.text');
-    let defaultConditionCodeErrorText = this.translate.instant('home.new-experiment.design.condition-name-validation.text')
+    let defaultConditionCodeErrorText = this.translate.instant('home.new-experiment.design.condition-name-validation.text');
     if (conditions.length >= 1 ) {
       const hasDefaultConditionCode = conditions.filter(
         condition => condition.conditionCode.toUpperCase() === defaultKeyword
       );
       if (!!hasDefaultConditionCode.length) {
-        this.conditionCodeError = defaultConditionCodeErrorText
+        this.conditionCodeError = defaultConditionCodeErrorText;
+      } else {
+        this.conditionCodeError = null;
+      }
+    }
+  }
+
+  validateHasAssignmentWeightsNegative(conditions: ExperimentCondition[]) {
+    let negativeAssignmentWeightErrorText = this.translate.instant('home.new-experiment.design.assignment-weight-negative.text');
+    if (conditions.length >= 1 ) {
```

Comment:
Can you update the if condition by just `if (conditions.length)`?

Reply:
ok